agent/structs/config_entry_gateways_test.go

```diff
@@ -562,3 +562,52 @@ func TestTerminatingConfigEntry_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestGatewayService_Addresses(t *testing.T) {
+	cases := []struct {
+		name     string
+		input    GatewayService
+		argument string
+		expected []string
+	}{
+		{
+			name:     "port is zero",
+			input:    GatewayService{},
+			expected: nil,
+		},
+		{
+			name: "no hosts",
+			input: GatewayService{
+				Port: 8080,
+			},
+			expected: []string{":8080"},
+		},
+		{
+			name: "no hosts with default",
+			input: GatewayService{
+				Port: 8080,
+			},
+			argument: "service.ingress.dc.domain",
+			expected: []string{"service.ingress.dc.domain:8080"},
+		},
+		{
+			name: "user-defined hosts",
+			input: GatewayService{
+				Port:  8080,
+				Hosts: []string{"*.test.example.com", "other.example.com"},
+			},
+			argument: "service.ingress.dc.domain",
+			expected: []string{"*.test.example.com:8080", "other.example.com:8080"},
+		},
+	}
+
+	for _, test := range cases {
+		// We explicitly copy the variable for the range statement so that can run
```

Comment:
Should the loop call `t.Parallel()`?

Reply:
I removed all the `t.Parallel` from `agent/structs` a little while ago (47238a693d96a335a4dc107d59e1905d32ae53f3) because the package runs tests in `0.02s` without parallel, and parallel tests cause problems with failure messages ( until go1.14.5/go1.15 fix it).  Even once it is fixed omitting it for packages that don't need it seems nice.

Reply:
I see, then we could remove the copy/comment.

Reply:
done